easybuild/easyblocks/m/mvapich2.py

```diff
@@ -134,10 +135,14 @@ def sanity_check_step(self):
         Custom sanity check for MVAPICH2
         """
         shlib_ext = get_shared_lib_ext()
-        custom_paths = {
-            'files': ['bin/%s' % x for x in ['mpicc', 'mpicxx', 'mpif77', 'mpif90', 'mpiexec.hydra']] +
-                     ['lib/lib%s' % y for x in ['fmpich', 'mpichcxx', 'mpichf90', 'mpich', 'mpl', 'opa']
-                                      for y in ['%s.a' % x, '%s.%s' % (x, shlib_ext)]],
-            'dirs': ['include'],
-        }
+        binaries = ['bin/%s' % x for x in ['mpicc', 'mpicxx', 'mpif77', 'mpif90', 'mpiexec.hydra']]
+        directories = ['include']
+        if LooseVersion(self.version) < LooseVersion('2.1'):
+            libraries = ['lib/lib%s' % y for x in ['fmpich', 'mpichcxx', 'mpichf90', 'mpich', 'mpl', 'opa']
+                                      for y in ['%s.a' % x, '%s.%s' % (x, shlib_ext)]]
+        else:
+            libraries = ['lib/lib%s' % y for x in ['mpi', 'mpicxx', 'mpifort']
+                                      for y in ['%s.a' % x, '%s.%s' % (x, shlib_ext)]]
```

Comment:
repeating the list comprehension feels a bit bloated, since only the library names are different

cleaned up version proposed in https://github.com/besserox/easybuild-easyblocks/pull/2